webclient/http1/src/main/java/io/helidon/webclient/http1/Http1CallChainBase.java

```diff
@@ -181,38 +213,21 @@ protected WebClientServiceResponse readResponse(WebClientServiceRequest serviceR
         ClientResponseHeaders responseHeaders = readHeaders(reader);
         connection.helidonSocket().log(LOGGER, TRACE, "client received headers %n%s", responseHeaders);
 
-        return createServiceResponse(serviceRequest, connection, reader, responseStatus, responseHeaders);
-    }
-
-    WebClientServiceResponse createServiceResponse(WebClientServiceRequest serviceRequest,
-                                                   ClientConnection connection,
-                                                   DataReader reader,
-                                                   Http.Status responseStatus,
-                                                   ClientResponseHeaders responseHeaders) {
-        WebClientServiceResponse.Builder builder = WebClientServiceResponse.builder();
-        AtomicReference<WebClientServiceResponse> response = new AtomicReference<>();
-
-        if (mayHaveEntity(responseStatus, responseHeaders)) {
-            // this may be an entity (if content length is set to zero, we know there is no entity)
-            builder.inputStream(inputStream(connection.helidonSocket(), response, responseHeaders, reader));
-        }
-
-        WebClientServiceResponse serviceResponse = builder
-                .connection(connection)
-                .headers(responseHeaders)
-                .status(responseStatus)
-                .whenComplete(whenComplete)
-                .serviceRequest(serviceRequest)
-                .build();
-
-        response.set(serviceResponse);
-        return serviceResponse;
+        return createServiceResponse(clientConfig,
+                                     serviceRequest,
+                                     connection,
+                                     reader,
+                                     responseStatus,
+                                     responseHeaders,
+                                     whenComplete);
     }
 
-    private InputStream inputStream(HelidonSocket helidonSocket,
-                                    AtomicReference<WebClientServiceResponse> response,
-                                    ClientResponseHeaders responseHeaders,
-                                    DataReader reader) {
+    private static InputStream inputStream(HttpClientConfig clientConfig,
```

Comment:
Since you already moved `createServiceResponse` because of it became static, I think you should move this as well.

Reply:
Not an issue after all.